CodeModel/src/main/java/org/openzen/zenscript/codemodel/compilation/impl/compiler/LocalTypeImpl.java

```diff
@@ -34,7 +34,7 @@ public Optional<TypeID> getSuperType() {
 
 	@Override
 	public StaticCallable thisCall() {
-		return resolvedThis.getConstructor().map(constructor -> new ThisCallable(thisType, constructor));
+		return resolver.resolve(thisType).getConstructor().map(constructor -> new ThisCallable(thisType, constructor));
```

Comment:
Resolving moved here because resolving earlier would break constructor delegation
```
public this() {
    this("hello");
}
```

Would not find any ctors unless resolved here.